### DISCUSSION TITLE
Why does compiler emit async state machine for async methods with only synchronous code?

### DISCUSSION THREAD

**Discussion by voroninp** [dotnet/csharplang#4088] [Q&A] [answered]
I can mark a method as async and do not use await, I get a warning from the compiler:
```CSharp
public async Task Foo()
{
}
```
When I look at IL (at least in LinqPad) I see that compiler emits code for state machine.

Why not just wrapping the result with the task? In this case it would be `return Task.CompltedTaks`

**Comment by YairHalberstadt** [comment, marked as answer]
This has been discussed on roslyn at https://github.com/dotnet/roslyn/issues/3659

In short, they would take this, but you would have to be careful to keep semantics exactly the same.
